src/transactions/utils.ts

```diff
@@ -47,3 +48,17 @@ export function groupFeedItemsInSections(
       data: value.data,
     }))
 }
+
+export const deduplicateTransactions = (
+  existingTxs: TokenTransaction[],
+  incomingTxs: TokenTransaction[]
+) => {
+  const currentHashes = new Set(existingTxs.map((tx) => tx.transactionHash))
+  const transactionsWithoutDuplicatedHash = existingTxs.concat(
+    incomingTxs.filter((tx) => !isEmpty(tx) && !currentHashes.has(tx.transactionHash))
```

Comment:
It looks like if incomingTxs has duplicates within itself then those will not be caught by this method. Is that a case we need to cover?

Reply:
that's a good point....i copied and paste this function so i didn't think too closely on this. if there are duplicates in `incomingTxs`, that should be a bug with blockchain-api which we can fix on the fly. the duplicate transaction would also be removed the next time this reducer is run (so i guess the impact of this issue depends on how often the user has new tx's)....i can add a filter to remove duplicates from the incoming txs

Reply:
i updated this @jh2oman :)